platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java

```diff
@@ -48,6 +82,123 @@ public User updateUserFromCodingcontest(ParticipationDto dto) {
         return userRepository.save(user);
     }
 
+    private Duration parseGameDuration(String duration){
+        LocalTime time = LocalTime.parse(duration);
+        int seconds = time.toSecondOfDay();
+        return Duration.ofSeconds(seconds);
+    }
+
+    private void createChallengeTemplate(CodingContestGameDto dto, UUID organizationId, Map<String, ByteArrayOutputStream> files) throws IOException {
+        ChallengeTemplate challengeTemplate = challengeTemplateRepository.findOneByCanonicalName(dto.getCanonicalName());
+        if (challengeTemplate != null){
+            return;
+        }
+        Organization organization = organizationRepository.findOne(organizationId);
+        if (organization == null){
+            throw new IllegalArgumentException("organization.invalid");
+        }
+
+        if (dto.getPuzzles()
+                .stream()
+                .findAny()
+                .filter(puzzleDto -> puzzleDto.getValidationClass() != null)
+                .isPresent()){
+            throw new IllegalArgumentException("ccc.game.structure.unsuported");
+        }
+
+        Endpoint taskEndpoint = endpointRepository.findOneByComponent("ccc-drone-task");
+        Endpoint challengeEndpoint = endpointRepository.findOneByComponent("ccc-challenge");
```

Comment:
Well I am a bit confused because this is hardcoded to `"ccc-drone-task"` .. .which sounds very much like "only valid for the newest CCC about drones".